### DISCUSSION TITLE
OMNeT++ 6.1.0 Released

### DISCUSSION THREAD

**Discussion by rhornig** [omnetpp/omnetpp#1231] [Announcements]
This update marks a significant refinement of OMNeT++ since version 6.0, impacting virtually every aspect of the simulation framework, including the IDE and simulation library. Among the various enhancements, the Analysis Tool stands out with major improvements to both its user interface and its underlying chart templates and Python library. For example, the new, highly customizable number and quantity formatting options facilitate the interpretation of raw simulation results displayed in the user interface; chart legends became much more customizable; and the enhanced plotting of enum-valued vectors allows much more appropriate visualization.

Additional notable updates include the ability to denote ini file sections as "abstract" (meaning they merely serve as base for further configurations); the validation of parameter values according to the `@enum` property; a new bookmarking feature and other enhancements in the Qtenv log window; a more streamlined operation of Project Features in the IDE; the possibility to have simulation programs dump the stack trace in the case of crashes; a new Python library for reading and querying NED files; and much more. [Here](https://github.com/omnetpp/omnetpp/blob/omnetpp-6.1.0/WHATSNEW.md) is a detailed summary of these key enhancements and other important changes in this release.

> [!IMPORTANT]
> Please read the installation instructions even if you are familiar with OMNeT++, as there are new dependencies and the recommended way of installing Python packages has also changed. There is a new `install.sh` script in the root of the source distribution that will help you install OMNeT++ on your machine.

<hr /><em>This discussion was created from the release <a href='https://github.com/omnetpp/omnetpp/releases/tag/omnetpp-6.1.0'>OMNeT++ 6.1.0</a>.</em>

**Comment by haug-den-lucas** [comment]
I cannot find the installation instructions for this release.

**Reply by rhornig** [reply]
The INSTALL.md file in the root is a good quick-start point.

**Comment by Lance729** [comment]
If I use wget to download, can I use the `omnetpp-latest-linux-x86_64.tgz` commend to download the latest version?

**Reply by rhornig** [reply]
no. each version has different version that is part of the filename. There is no 'latest' version. (opp_env does have omnetpp-latest but that is an extra indirection over the filenames and the opp_env database itself knows which is the latest version)

**Comment by kruviser** [comment]
Hi Rudolf, when installing omnet 6.1.0 on MAC OS (AARCH64) I believe users have to choose a directory whose full path does not contain any space (using a path containing spaces gave me problem indeed). This is similar to what you mentioned in the installation guide for windows OS, but it is not mentioned in the MAC OS section.

**Reply by rhornig** [reply]
Actually it is true for ALL operating systems. It is not windows specific. Also, the 'setenv' script checks whether the path contains a space and throws an error. As you are supposed to start with sourcing 'setenv', you should not be able to proceed with an incorrect path.

ps: I'm not sure how this plays with the new `install.sh` script which sources `setenv` only twoards the end of the install process. Still, it must fail with an install path with a space.

It was mentioned in Windows install guide only because it was more common on windows to run into these issues (with common directories like `Program Files`, `Documents and Settings` etc.)

**Comment by fritz-42** [comment]
I've some issues with the installation of the latest package.
Details in the topic: https://github.com/omnetpp/omnetpp/discussions/1366
